PowerUp/app/src/main/java/powerup/systers/com/FinalAvatarActivity.java

```diff
@@ -28,73 +24,33 @@ public void onCreate(Bundle savedInstanceState) {
-        setmDbHandler(new DatabaseHandler(this));
-        getmDbHandler().open();
+        dataSource = InjectionClass.provideDataSource(this);
 
         // initialize variables
         setContentView(R.layout.final_avatar);
-        eyeAvatar = (ImageView) findViewById(R.id.eye_view);
-        hairAvatar = (ImageView) findViewById(R.id.hair_view);
-        skinAvatar = (ImageView) findViewById(R.id.skin_view);
-        clothAvatar = (ImageView) findViewById(R.id.dress_view);
+        ImageView eyeAvatar = findViewById(R.id.eye_view);
+        ImageView hairAvatar = findViewById(R.id.hair_view);
+        ImageView skinAvatar = findViewById(R.id.skin_view);
+        ImageView clothAvatar = findViewById(R.id.dress_view);
 
-        // extract selected eyes,skin,cloth,hair from database & apply it on avatar
-        String eyeImageName = getResources().getString(R.string.eye);
-        eyeImageName = eyeImageName + getmDbHandler().getAvatarEye();
-        R.drawable ourRID = new R.drawable();
-        java.lang.reflect.Field photoNameField;
-        try {
-            photoNameField = ourRID.getClass().getField(eyeImageName);
-            eyeAvatar.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
-
-        String skinImageName = getResources().getString(R.string.skin);
-        skinImageName = skinImageName + getmDbHandler().getAvatarSkin();
-        try {
-            photoNameField = ourRID.getClass().getField(skinImageName);
-            skinAvatar.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
-
-        String clothImageName = getResources().getString(R.string.cloth);
-        clothImageName = clothImageName + getmDbHandler().getAvatarCloth();
-        try {
-            photoNameField = ourRID.getClass().getField(clothImageName);
-            clothAvatar.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
-
-        String hairImageName = getResources().getString(R.string.hair);
-        hairImageName = hairImageName + getmDbHandler().getAvatarHair();
-        try {
-            photoNameField = ourRID.getClass().getField(hairImageName);
-            hairAvatar.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException error) {
-            error.printStackTrace();
-        }
+        // use DbResourceUtil to extract data from database & apply it on avatar
+        DbResourceUtils resourceUtils = new DbResourceUtils(dataSource, FinalAvatarActivity.this);

```

Comment:
Its good to check, if the object is created or is null, and then make any method call on that object.